Make IsActivated property public in Window.cs

pictos opened the pull request:
Making this public will allow devs to filter and find the `Window` that has focus, it can make easier to show popups and perform some actions.


Related to #30762

Make IsActivated property public in Window.cs

The `IsActivated` property in the `Microsoft.Maui.Controls` namespace
was changed from `internal` to `public`. This increases its visibility,
allowing it to be accessed from outside the assembly.

pictos committed:
Make IsActivated property public in Window.cs

The `IsActivated` property in the `Microsoft.Maui.Controls` namespace
was changed from `internal` to `public`. This increases its visibility,
allowing it to be accessed from outside the assembly.

pictos committed:
update public api txt

mattleibow reviewed (changes requested):
For public properties, this really should be a bindable readonly property

Also

Can you add somrthing to the mutli-window sample page
* a binding to a label on that.

And maybe update/duplicate the tests here:

* src/Controls/tests/Core.UnitTests/WindowsTests.cs
* src/Controls/tests/DeviceTests/Elements/Window/WindowTests.Windows.cs



This is the wondow property on VisualElement, so ignore the name:

```cs
static readonly BindablePropertyKey WindowPropertyKey = BindableProperty.CreateReadOnly(
	nameof(Window), typeof(Window), typeof(VisualElement), null, propertyChanged: OnWindowChanged);

/// <summary>Bindable property for <see cref="Window"/>.</summary>
public static readonly BindableProperty WindowProperty = WindowPropertyKey.BindableProperty;

/// <summary>
/// Gets the <see cref="Window"/> that is associated with an element. This is a read-only bindable property.
/// </summary>
public Window Window => (Window)GetValue(WindowProperty);

/// <inheritdoc/>
Window IWindowController.Window
{
	get => (Window)GetValue(WindowProperty);
	set => SetValue(WindowPropertyKey, value);
}
```

I would expect something like this:

```cs
static readonly BindablePropertyKey IsActivatedPropertyKey = BindableProperty.CreateReadOnly(
	nameof(IsActivated), typeof(bool), typeof(Window), false);
public static readonly BindableProperty IsActivatedProperty = IsActivatedPropertyKey.BindableProperty;
public bool IsActivated
{
    get => (bool)GetValue(IsActivatedProperty)
    private set => SetValue(IsActivatedPropertyKey, value);
}
```

Also, we will need to remove the `_isActivated` field in window.cs.

pictos committed:
Window modifications

pictos committed:
public apis and fix code

pictos committed:
Make Window.IsActivated property read-only

Changed Window.IsActivated to use a read-only BindableProperty and made its setter private. This restricts modification of the activation state to internal code, improving encapsulation and preventing external changes.

pictos committed:
Update IsActivated property and add window activation tests

Changed Window.IsActivated setter to use IsActivatedPropertyKey for proper encapsulation. Added unit tests to verify window activation and deactivation behavior, including Appearing event handling. Included new using directives for async test support.

jfversluis commented:
/azp run

pictos commented:
BTW, is still missing DeviceTests, will implement it this week

mattleibow commented:
This is great! Slowly bindable property-izing he world.

This will be nice to do binding to IsActivated.

Can you maybe also add a test for that using a binding to that property. Just to make sure that it does work.

rmarinho reviewed (commented):
Possible to add a DeviceTest ? clean up. the comment out code.

pictos committed:
Add test to make sure Matthew and Rui are super happy 🙌

mattleibow committed:
Revert unnecessary diff

mattleibow commented:
/rebase

mattleibow reviewed (approved):
Matthew is super happy!
